### PR TITLE
Fixing distro deprecation in setup.py

### PR DESCRIPTION
**Pull request by azzamh15**
fixed distro depreciated features and made not implemented error clear

### CODE REVIEW

File: PythonAPI/carla/setup.py [hunk 1]
```diff
@@ -34,8 +34,9 @@ def walk(folder, file_filter='*'):
     if os.name == "posix":
         import distro
 
-        linux_distro = distro.linux_distribution()[0]
-        if linux_distro.lower() in ["ubuntu", "debian", "deepin"]:
+        supported = ["ubuntu", "debian", "deepin"]
```
Comment (xavisolesoft) [inline PythonAPI/carla/setup.py:37]:
```suggestion
        supported_dists = ["ubuntu", "debian", "deepin"]
```

This variable have a different name on the `NotImplementedError` Exception message.

**Review by xavisolesoft — CHANGES_REQUESTED** [review]

**Review by azzamh15 — COMMENTED** [review]
oh really sorry

**Comment by Daraan** [comment]
For reference this is duplicate of #6802

**Comment by azzamh15** [comment]
> For reference this is duplicate of #6802

yes merging branch was stale

**Review by xavisolesoft — CHANGES_REQUESTED** [review]
Duplicated

**Comment by azzamh15** [comment]
closing this pull request as duplicate of #6802

**Pull request closed without merging**